--- objects/skaters/goalie.gd
+++ objects/skaters/goalie.gd
@@ -35,14 +35,13 @@
 func _physics_process(delta: float) -> void:
	if ghost:
		ghost.handle(delta, self)
-	else:
-		var diffx = global_position.x - home_x
-		if abs(diffx) > 4:
-			apply_impulse(Vector2.LEFT * diffx)
-		var puck_node = Globals.get_closest_node(global_position, "pucks")
-		if puck_node:
-			var diffy = global_position.y - clamp(puck_node.global_position.y + OFFSET, min_y, max_y)
-			if abs(diffy) > 4:
-				apply_impulse(Vector2.UP * diffy)
-			apply_impulse(Vector2.UP * diffy)
+		return
+	var diffx = global_position.x - home_x
+	if abs(diffx) > 4:
+		apply_impulse(Vector2.LEFT * diffx)
+	var puck_node = Globals.get_closest_node(global_position, "pucks")
+	if puck_node:
+		var diffy = global_position.y - clamp(puck_node.global_position.y + OFFSET, min_y, max_y)
+		if abs(diffy) > 4:
+			apply_impulse(Vector2.UP * diffy)
	#counter += 1
